public/app/features/alerting/unified/components/contact-points/useNotificationTemplates.ts

```diff
@@ -0,0 +1,138 @@
+import { produce } from 'immer';
+
+import { ActiveTab as ContactPointsActiveTabs } from 'app/features/alerting/unified/components/contact-points/ContactPoints';
+import { useDispatch } from 'app/types';
+
+import { AlertManagerCortexConfig } from '../../../../../plugins/datasource/alertmanager/types';
+import { alertmanagerApi } from '../../api/alertmanagerApi';
+import { updateAlertManagerConfigAction } from '../../state/actions';
+import { ensureDefine } from '../../utils/templates';
+import { TemplateFormValues } from '../receivers/TemplateForm';
+
+// TODO Check the issue with useAlertmanagerConfig hook
+export function useNotificationTemplates(amSourceName?: string) {
+  const templatesRequestState = alertmanagerApi.endpoints.getAlertmanagerConfiguration.useQuery(amSourceName || '', {
+    skip: !amSourceName,
+    selectFromResult: (state) => ({
+      ...state,
+      data: state.data ? amConfigToTemplates(state.data) : undefined,
+      currentData: state.currentData ? amConfigToTemplates(state.currentData) : undefined,
+    }),
+  });
+
+  return templatesRequestState;
+}
+
+export interface NotificationTemplate {
+  name: string;
+  template: string;
+  provenance?: string;
+}
+
+function amConfigToTemplates(config: AlertManagerCortexConfig): NotificationTemplate[] {
+  return Object.entries(config.template_files).map(([name, template]) => ({
+    name,
+    template,
+    provenance: (config.template_file_provenances ?? {})[name],
```

Comment:
Can you make it fall back to `PROVENANCE_NONE` constant (its in mute timings file at the moment)? As the k8s API will return that value for non-provisioned entities. Then we should change the check to be `!== PROVENANCE_NONE` in the appropriate places